src/main.c

```diff
@@ -1712,6 +1713,7 @@ GOptionEntry entries_install[] = {
 
 GOptionEntry entries_bundle[] = {
 	{"signing-keyring", '\0', 0, G_OPTION_ARG_FILENAME, &signing_keyring, "verification keyring file", "PEMFILE"},
+	{"squashfs-args", '\0', 0, G_OPTION_ARG_STRING_ARRAY, &squashfs_args, "squashfs extra args (repeatable)", ""},
```

Comment:
All other options note a placeholder for the content to add, this should be done here, too for consistency reasons. Thus maybe

```suggestion
	{"squashfs-args", '\0', 0, G_OPTION_ARG_STRING_ARRAY, &squashfs_args, "squashfs extra args (repeatable)", "ARGS"},
```

Reply:
How is this used for mksquashfs args which have an option?
`--squashfs-args "-processors 8"` or `--squashfs-args -processors --squashfs-args 8`

Often, programs expect that the shell has split their arguments into separate argv strings and fail when options values are part of the same argv string as the option itself.

glib has https://developer.gnome.org/glib/stable/glib-Shell-related-Utilities.html#g-shell-parse-argv, so an alternative would be to supply one string argument to rauc and let it split that up for the mksquashfs call. Then you could pass `--squashfs-args "-processors 8 -comp xz"` and it would work independent of clever handling in mksquashfs.

Reply:
> How is this used for mksquashfs args which have an option?
> `--squashfs-args "-processors 8"` or `--squashfs-args -processors --squashfs-args 8`
> 
> Often, programs expect that the shell has split their arguments into separate argv strings and fail when options values are part of the same argv string as the option itself.
> 
Generally: `--squashfs-args "-processors" --squashfs-args "8"` would work - I agree it's a bit ugly though.

> glib has https://developer.gnome.org/glib/stable/glib-Shell-related-Utilities.html#g-shell-parse-argv, so an alternative would be to supply one string argument to rauc and let it split that up for the mksquashfs call. Then you could pass `--squashfs-args "-processors 8 -comp xz"` and it would work independent of clever handling in mksquashfs.
I'll give it a go, agreed this would be cleaner - not particuarly familiar with glib, so thanks for pointing this out :)

Reply:
This worked a treat, thanks again. Updated PR.